src/core/templates/show-logo.js

```diff
@@ -1,35 +1,43 @@
 // @ts-check
+import { docLink, showError } from "../../core/utils.js";
 import { html } from "../../core/import-maps.js";
-import { showWarning } from "../../core/utils.js";
 
 const name = "core/templates/show-logo";
 
 /**
+ * Logo mapper. Takes a logo structure and converts it to HTML.
+ *
  * @param {object} logo
  * @param {string} logo.src
  * @param {string} logo.url
  * @param {string} logo.alt
  * @param {string} [logo.id]
  * @param {number} [logo.width]
  * @param {number} [logo.height]
+ * @param {number} index
  */
-export default function showLogo(logo) {
+export default function showLogo(logo, index) {
   /** @type {HTMLAnchorElement} */
-  const a = html`<a href="${logo.url || ""}" class="logo"></a>`;
-  if (!logo.alt) {
-    const msg = "Found spec logo without an `alt` attribute.";
-    showWarning(msg, name, { elements: [a] });
+  const a = html`<a href="${logo.url || null}" class="logo"
+    ><img
+      alt="${logo.alt || null}"
+      height="${logo.height || null}"
+      id="${logo.id || null}"
+      width="${logo.width || null}"
+      src="${logo.src || null}"
+      crossorigin
+    />
+  </a>`;
+  const img = a.querySelector("img");
+  if (!img.alt) {
+    const msg = `Logo at index ${index} is missing required "\`alt\`" property.`;
```

Comment:
If we've src available here, maybe we can show that in error msg instead of index?

Reply:
I'll show both... why compromise! :D